### PR TITLE
Added Profiler Context to manage all profilers

### PR DESCRIPTION
**Pull request by 1SAA**

### CODE REVIEW

File: tests/test_profiler/test_comm_prof.py [hunk 1]
```diff
@@ -4,11 +4,11 @@
...
-from colossalai.utils.profiler import enable_communication_prof, communication_prof_show
+from colossalai.utils.profiler import CommProfiler, ProfilerContext
 
 BATCH_SIZE = 1024
 D_MODEL = 1024
-CONFIG = dict(parallel=dict(tensor=dict(mode='1d', size=4)))
+CONFIG = dict()
 
 
 def run_test(rank, world_size, port):
```
Comment (feifeibear) [inline tests/test_profiler/test_comm_prof.py:14]: This is not a valid test. You only make sure the profiler does not fail.

File: colossalai/utils/profiler/comm_profiler.py [hunk 2]
```diff
@@ -101,6 +111,30 @@ def reset(self):
         self.pending_metadata = None
         self.warn_flag = False
 
+    def enable(self):
+        global COL_COMM_PROF
+        COL_COMM_PROF = self
```
Comment (FrankLeeeee) [inline colossalai/utils/profiler/comm_profiler.py:116]: seems strange to me, is this a better way to write this line?

```python
def all_reduce(profiler: CommProfier,
                    tensor: torch.Tensor,
                    op: ReduceOp = ReduceOp.SUM,
                    group=None,
                    async_op: bool = False,
                         ):
        ...


dist.all_reduce = functools.partial(all_reduce, profiler=self)

```

**Review by feifeibear — COMMENTED** [review]
Add a readme for profiler usage.
The unittest is a not valid one.

**Comment by FrankLeeeee** [comment]
What is the relationship between this PR and #329 ? @1SAA @kurisusnowdeng I see this PR as a replacement for #329 .

File: colossalai/utils/profiler/prof_utils.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+class ProfilerContext(object):
+    """
+    Profiler context manager
+    Usage:
+    from colossalai.utils.profiler import ProfilerActivity, profiler
+    from torch.utils.tensorboard import SummaryWriter
+    with profiler(activities=[ProfilerActivity.COMMUNICATION]) as prof:
+        train()
+    writer = SummaryWriter('tb/path')
```
Comment (FrankLeeeee) [inline colossalai/utils/profiler/prof_utils.py:42]: This is not consistent with `__init__`

**Comment by kurisusnowdeng** [comment]
@1SAA Remember to keep update codes here with #329

File: colossalai/utils/profiler/comm_profiler.py (hunk 2 above)
Comment (1SAA) [reply colossalai/utils/profiler/comm_profiler.py]: Done

File: tests/test_profiler/test_comm_prof.py (hunk 1 above)
Comment (1SAA) [reply tests/test_profiler/test_comm_prof.py]: It has been deleted.

File: colossalai/utils/profiler/prof_utils.py (hunk 3 above)
Comment (1SAA) [reply colossalai/utils/profiler/prof_utils.py]: Done.

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**